napari/layers/vectors/vectors.py

```diff
@@ -593,15 +617,64 @@ def edge_contrast_limits(
     @property
     def _view_face_color(self) -> np.ndarray:
         """(Mx4) np.ndarray : colors for the M in view vectors"""
-        face_color = np.repeat(self.edge_color[self._view_indices], 2, axis=0)
+        face_color = self.edge_color[self._view_indices]
+        face_color[:, -1] *= self._view_alphas
+        face_color = np.repeat(face_color, 2, axis=0)
+
         if self._ndisplay == 3 and self.ndim > 2:
             face_color = np.vstack([face_color, face_color])
 
         return face_color
 
+    def _slice_data(
+        self, dims_indices
+    ) -> Tuple[List[int], Union[float, np.ndarray]]:
+        """Determines the slice of vectors given the indices.
+
+        Parameters
+        ----------
+        dims_indices : sequence of int or slice
+            Indices to slice with.
+
+        Returns
+        -------
+        slice_indices : list
+            Indices of vectors in the currently viewed slice.
+        alpha : float, (N, ) array
+            If in `n_dimensional` mode then the opacity of vectors, where
```

Comment:
```suggestion
            If in `n_dimensional` mode then the opacity of vectors in the current slice, where
```

will probably break formatting if merged as is